Update to 1.21.1

cpaca opened the pull request.

cpaca committed:
Update to 1.21 - WARNING: DOESN'T WORK

ProtocolLib note: 5.2.0-SNAPSHOT wasn't working, so I just made it 5.4.0.

I don't know what ProtocolLib is being used for, so I'm leaving it at this until we put more thought into it.

Future commits will be about fixing all of the bugs (some things got removed/deprecated.)

cpaca committed:
Remove EnchantmentRarity

"As of 1.20.5 enchantments do not have a rarity."

https://jd.papermc.io/paper/1.21.8/org/bukkit/enchantments/Enchantment.html

cpaca committed:
getActiveSlots is deprecated.

It's been replaced with getActiveSlotGroups. From what I can tell, the function isn't used anyway, but it's here.

I'm getting the vibe I should delete the whole file.

cpaca committed:
Revamp EnchantmentListener

cpaca committed:
Reset CustomEnchantment to scratch

I might have removed too much, but from what I can tell, everything I removed was just added because it was required by the old class.

I kept getCostMultiplier as it's used by AnvilListener, and lore as it's not @Override.

cpaca committed:
Nuking AnvilListener

Not sure what it needs yet, but I do think it still needs to exist.

cpaca committed:
Update Enchantable to use minecraft registries.

cpaca committed:
Remove EnchantManager

cpaca committed:
Delete EnchantCommand.java

cpaca committed:
Fix accidentally-deleted semicolon

cpaca committed:
Fix CustomLootRegistry Enchantments

cpaca committed:
CustomEnchantment implements EnchantmentAbility

cpaca committed:
CustomEnchantment implements EnchantmentAbility

cpaca committed:
Merge branch '1.21-update' of https://github.com/cpaca/HoloItems into 1.21-update

cpaca committed:
Remove now-extraneous functions

These will be defined by the datapack.

cpaca committed:
Remove Tide Rider particles

cpaca committed:
Update John Rengelman's Shadow

It's now managed by GradleUp.

cpaca committed:
Fix enchantment namespace

cpaca committed:
have to use holoitems namespace

enchantable items are getting their key using CustomItem's getKey(), which uses the plugin.
This will be resolved in a future version, as Enchantable only exists on CustomItem objects, which (now) only exist to add one recipe each.

cpaca committed:
Register and listen to enchantments

cpaca committed:
Re-register enchantments again

cpaca committed:
Replace unstackable with integer stackSize

cpaca committed:
Merge branch '1.21-update' of https://github.com/cpaca/HoloItems into 1.21-update

cpaca committed:
Add datapack and associated gradle task

cpaca committed:
Make copyDatapack delete old datapack

cpaca committed:
Fix enchantment namespace

cpaca committed:
Move Backdash recipe to datapack

Doing this one early because I need it for another commit I wanna try.

cpaca committed:
Implement book-like items

Also makes backdash booklike.

cpaca committed:
Remove Keys.OWNER

dlee13 commented on build.gradle.kts:
```diff
@@ -64,15 +64,28 @@ tasks {
     runServer {
-        minecraftVersion("1.20.1")
+        dependsOn("copyDatapack")
+        minecraftVersion("1.21")
```
```suggestion
        minecraftVersion("1.21.1")
```

dlee13 commented on build.gradle.kts:
```diff
@@ -64,15 +64,28 @@ tasks {
 bukkit {
     main = "xyz.holocons.mc.holoitemsrevamp.HoloItemsRevamp"
-    apiVersion = "1.20"
+    apiVersion = "1.21"
```
```suggestion
    apiVersion = "1.21.1"
```

cpaca committed:
Grindstone now removes BookLike

cpaca committed:
Make BookLike a boolean

cpaca committed:
make non-booklike anvil uses work again

cpaca committed:
Switch to item_name instead of custom_name

cpaca committed:
Fix acquire command

cpaca committed:
Move all recipes to datapack

cpaca committed:
Simplify/reduce code.

This is equivalent to the old code, since there are now no more implementations of getRecipe.

This is to help justify the next commit (which is separate, because it does a lot and I might need to undo it.)

cpaca committed:
Remove RecipeBuilder Shaped, Shapeless, Furnace

Advanced is something I need to look into and decide if I want to keep it or completely rebuild it.

These things make no sense to keep though, due to datapack stuff.

cpaca committed:
Correct runServer version

Co-authored-by: Derek Lee <derek.evilcornrules@gmail.com>

cpaca committed:
Correct apiVersion version

Co-authored-by: Derek Lee <derek.evilcornrules@gmail.com>

cpaca committed:
Switch from bukkit plugin.yml to paper-plugin.yml

MainCommand is now broken, as command registration is different with paper-plugin.yml

dlee13 commented on build.gradle.kts:
```diff
@@ -1,6 +1,7 @@
 plugins {
     `java-library`
-    id("net.minecrell.plugin-yml.bukkit") version "0.6.0" // Generates plugin.yml
+    id("java")
```
Is this needed?

cpaca replied:
See other comment

dlee13 commented on build.gradle.kts:
```diff
@@ -18,6 +19,7 @@ repositories {
     maven("https://repo.papermc.io/repository/maven-public/")
     maven("https://repo.dmulloy2.net/repository/public/")
     maven("https://maven.enginehub.org/repo/")
+    maven("https://papermc.io/repo/repository/maven-public/")
```
Why was this added?

cpaca replied:
IDE was saying "code doesn't work" so I eventually resorted to following the installation directions by-the-letter. 
Forgot to remove the extraneous stuff after fixing my IDE.

dlee13 commented on src/main/java/xyz/holocons/mc/holoitemsrevamp/HoloitemsBootstrap.java:
```diff
@@ -0,0 +1,18 @@
+import org.jetbrains.annotations.NotNull;
+import xyz.holocons.mc.holoitemsrevamp.command.MainCommand;
+
+public class HoloitemsBootstrap implements PluginBootstrap {
```
Fix filename

cpaca replied:
I assume you mean HoloItems (capital I). 
If you do: Fixed.

cpaca committed:
Fix command

cpaca committed:
Remove CraftListener, RecipeBuilder, RecipeManager.

cpaca committed:
Remove extraneous recipe classes

cpaca committed:
Add CustomItem.editRecipe

cpaca committed:
Update dependency (InventoryFramework)

cpaca committed:
Fix bootstrap name

cpaca committed:
Fix bootstrap name

cpaca committed:
Merge branch '1.21-update' of https://github.com/cpaca/HoloItems into 1.21-update

cpaca committed:
Remove unnecessary build.gradle.kts lines

cpaca committed:
Scrapping bootstrap idea.

I'm leaving the file for now, though.

cpaca committed:
Remove Keys.UNSTACKABLE

cpaca committed:
Remove CustomItem.updateStack

I think this was made so that if an ItemStack ended up "malformed" somehow this could fix it?

That's about the only explanation I can think of.

cpaca committed:
Move HolyFire recipe fully back into code

Yeah, he's right. Having two different places which BOTH need to be loaded for the recipe to load properly is really dumb.

cpaca committed:
Add and shade lightbend (hocon)

cpaca committed:
Move recipes back to code format

Since we can't have all of them in the datapack, they are going to be all in the code.

cpaca committed:
remove extraneous tag

cpaca committed:
Make Backdash booklike again

cpaca committed:
add holoitems:enchantments tag

cpaca committed:
Fix TideRider recipe

cpaca committed:
Rename tag enchantments -> holoenchantments

cpaca committed:
Remove outdated CustomEnchantment functions

cpaca committed:
Remove ProtocolLib dependency

No longer being used.

cpaca committed:
remove old todo

cpaca committed:
Fix TideRider's particle

Spawning 80 seems like too much though.

cpaca committed:
Attempt to fix GitHub workflow

cpaca committed:
Update remaining build.yml versions

cpaca committed:
Fix using holoitems in non-custom recipes.

cpaca committed:
Actually register recipes

cpaca committed:
Null-check for recipe registration

cpaca committed:
Fix minRow/minCol calculations

cpaca committed:
Make getRecipe protected again

Apparantly I changed the privacy level when I reimplemented the recipe

cpaca committed:
Use HoloEnchantment tag instead of namespace check

cpaca committed:
Explain BookLike try/catch clause

cpaca committed:
Fix tag lookup

cpaca committed:
Move enchantment tag checks to a manager

Idea: What if we deleted the CustomEnchantment class?

cpaca committed:
Remove CustomEnchantment

cpaca committed:
Suppress UnstableApiUsage for lifecycle API

cpaca committed:
Make EnchantmentAbility not Keyed

cpaca committed:
Register enchantments in Integrations

cpaca committed:
Rename loadCustomEnchantments

cpaca committed:
Use better entry construction in EnchantManager

cpaca committed:
Fix javadoc error (param name was changed)

cpaca committed:
Remove old command registration
